wai-aria/scripts/aria-utils.js

```diff
@@ -0,0 +1,61 @@
+/* Utilities related to WAI-ARIA */
+
+const AriaUtils = {
+
+  /*
+  Tests simple role assignment: <div role="alert">x</div>
+  Not intended for nested, context-dependent, or other complex role tests.
+  */
+  assignAndVerifyRolesByRoleNames: function(roleNames) {
+    for (const role of roleNames) {
+      promise_test(async t => {
+        let el = document.createElement("div");
+        el.appendChild(document.createTextNode("x"));
+        el.role = role;
```

Comment:
`role` IDL attribute isn't implemented in Gecko yet. It doesn't seem necessary to rely on it here.
```suggestion
        el.setAttribute("role", role);
```

Reply:
```suggestion
        el.setAttribute("role", role); // el.role not yet supported by Gecko.
```